File: app/src/main/java/com/skyline/app/TeamAdapter.java
package com.skyline.app;

import android.view.LayoutInflater;
import android.view.ViewGroup;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.skyline.app.databinding.ItemTeamBinding;
import com.skyline.model.TeamMember;
import java.util.List;

public class TeamAdapter extends RecyclerView.Adapter<TeamAdapter.TeamViewHolder> {
    private List<TeamMember> items;

    public TeamAdapter(List<TeamMember> items) {
        this.items = items;
    }

    @NonNull
    @Override
    public TeamViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        ItemTeamBinding binding = ItemTeamBinding.inflate(LayoutInflater.from(parent.getContext()), parent, false);
        return new TeamViewHolder(binding);
    }

    @Override
    public void onBindViewHolder(@NonNull TeamViewHolder holder, int position) {
        holder.bind(items.get(position % items.size()));
    }

    @Override
    public int getItemCount() {
        return items == null || items.isEmpty() ? 0 : Integer.MAX_VALUE;
    }

    static class TeamViewHolder extends RecyclerView.ViewHolder {
        private ItemTeamBinding binding;

        public TeamViewHolder(ItemTeamBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }

        public void bind(TeamMember item) {
            binding.imgTeam.setImageResource(item.getImageRes());
            binding.tvTeamName.setText(item.getName());
        }
    }
}

File: app/src/main/java/com/skyline/app/TeamAdapter.kt
package com.skyline.app

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.skyline.app.databinding.ItemTeamBinding
import com.skyline.model.TeamMember

class TeamAdapter(
    private val items: List<TeamMember>
) : RecyclerView.Adapter<TeamAdapter.TeamViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): TeamViewHolder {
        val binding = ItemTeamBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return TeamViewHolder(binding)
    }

    override fun onBindViewHolder(holder: TeamViewHolder, position: Int) {
        holder.bind(items[position])
    }

    override fun getItemCount(): Int = items.size

    class TeamViewHolder(
        private val binding: ItemTeamBinding
    ) : RecyclerView.ViewHolder(binding.root) {
        fun bind(item: TeamMember) = with(binding) {
            imgTeam.setImageResource(item.imageRes)
            tvTeamName.text = item.name
        }
    }
}
